### PR TITLE
Update test class prefix

### LINKED ISSUE #1245: Update test prefix in test files and classes
As @westonruter mentioned in https://github.com/WordPress/performance/issues/514#issuecomment-2129970271, we should maintain consistency in the naming of test classes and files by adding the prefixes `Test_` for classes and `test-` for files. This approach aligns with the naming conventions of core test files and classes.

### PR DESCRIPTION
**Pull request by sarthak-19**
Fixes #1245

## Summary

Update prefix `test-` in files and `Test_` in class names in the following plugin tests:
- auto-sizes
- dominant-color-images
- embed-optimizer
- optimization-detective
- performance-lab
- speculation-rule
- webp-uploads

## Relevant technical choices

N/A

### CODE REVIEW

**Comment by sarthak-19** [comment]
@thelovekesh Checks failing due to missing labels.

File: tests/plugins/optimization-detective/storage/rest-api-tests.php
Comment (westonruter) [inline tests/plugins/optimization-detective/storage/rest-api-tests.php:1]: Looks like this file needs to be renamed to `test-rest-api.php`

File: tests/plugins/optimization-detective/storage/data-tests.php
Comment (westonruter) [inline tests/plugins/optimization-detective/storage/data-tests.php:1]: Rename to `test-data.php`?

File: tests/plugins/optimization-detective/storage/class-od-url-metrics-post-type-tests.php
Comment (westonruter) [inline tests/plugins/optimization-detective/storage/class-od-url-metrics-post-type-tests.php:1]: Rename file to prefix with `test-`?

File: tests/plugins/optimization-detective/storage/class-od-storage-lock-tests.php
Comment (westonruter) [inline tests/plugins/optimization-detective/storage/class-od-storage-lock-tests.php:1]: Rename file to prefix with `test-`?

**Review by westonruter — CHANGES_REQUESTED** [review]

File: tests/plugins/optimization-detective/storage/rest-api-tests.php
Comment (sarthak-19) [reply tests/plugins/optimization-detective/storage/rest-api-tests.php:1]: Renamed the missing files

File: tests/plugins/performance-lab/includes/site-health/audit-enqueued-assets/testaudit-enqueued-assets.php
Comment (westonruter) [inline tests/plugins/performance-lab/includes/site-health/audit-enqueued-assets/testaudit-enqueued-assets.php:1]: Should be `test-audit-enqueued-assets.php`

**Comment by westonruter** [comment]
Let's also rename `tests/plugins/optimization-detective/uninstall-tests.php` to `test-uninstall.php`

**Review by westonruter — APPROVED** [review]

**Pull request merged**